text/1001-edgedb-install.rst

```diff
@@ -0,0 +1,298 @@
+  EdgeDB server or all versions of EdgeDB from the local machine;
+
+* ``edgedb server start`` -- starts an EdgeDB server instance;
+
+* ``edgedb server status`` -- show the status of the local EdgeDB server;
+
+* ``edgedb server stop`` -- stop the given EdgeDB server instance;
+
+* ``edgedb server upgrade`` -- upgrade the specified EdgeDB server instance
+  to the new major version.
+
+* ``edgedb server prune`` -- removes upgrade backups and other unused data.
+
+Common aspects of the implementation of the above commands is discussed
+below in `design-considerations`_.
+
+
+edgedb server install
+=====================
+
+Downloads and installs a given EdgeDB server version
+(latest stable by default).
+
+Arguments
+---------
+
+``--version=<ver>``
+  specifies the major version of the server to install.
+
+``--nightly``
+  if passed, the latest nightly build from the specified version channel
+  is installed.
+
+``--user``
+  install into the user home directory instead of system-wide, does not
+  require privelege elevation (TODO: this requires relocatable
+  builds and is TBD).
+
+``--list``
+  list EdgeDB versions available for installation
+
+``--list-installed``
+  list installed EdgeDB versions
+
+``--update``
+  if specified, ``edgedb install`` will only attempt to update the existing
+  installations.
+
+``--docker``
+  Use Docker instead of downloading and installing packages directly onto
+  the user's system.  The Docker daemon must be present and accessible by
+  the user.
+
+
+Implementation
+--------------
+
+By default, ``edgedb install`` will look for the most suitable package for
+the current platform and install it using the system's package manager
+if one is available, or download and unpack a generic package for the
+OS/arch, or use Docker if avalable (and requested).
+
+
+edgedb server uninstall
+=======================
+
+Uninstalls the specified version of EdgeDB.
+
+Synopsis
+--------
+
+``edgedb server uninstall [options]``
+
+If there are multiple versions installed, either ``--all`` or
+``--version`` or ``--unused`` is required.
+
+Options
+-------
+
+``--version=<ver>``
+  Specifies the version to uninstall.  The specified server version must
+  not be currently running.
+
+``--all``
+  Uninstalls all versions of EdgeDB.
+
+``--unused``
+  Uninstalls all versions of EdgeDB that are not used in any instance.
+
+
+edgedb server start
+===================
+
+Starts an EdgeDB server instance with the specified name.
+
+Synopsis
+--------
+
+``edgedb server start [options] [<name>]``
+
+Options
+-------
+
+``<name>``
+  The name of the EdgeDB instance.  Must be unique.  If not specified,
+  the name ``default`` is used.
+
+``--version=<ver>``
+  Optionally specifies the server version to use.  If not specified,
+  the latest installed server version is used.
+
+``--start-conf=auto|manual``
+  If set to ``auto`` (the default), the server will be started automatically
+  on system boot.
+
+``--port=<port-number>``
```

Comment:
This is a bit weird. I would expect port to be remembered when I create an instance and then when I start it without remembering the port.

Since we have single command which is "create if not exists" it's unclear whether I must specify non-default port on subsequent start and whether it should fail (to show mistake) or silently update port in case it different.

Reply:
Yes. The port is remembered in the configuration.  This `--port` is to explicitly start the server on a different port.